### PR TITLE
Quotify initial project setup

### PR DESCRIPTION
**Pull request by itmani1991**
# Pull Request

## Description

Quotify initial project setup with initial UI plan on paper and UI in jetpack compose

## Checklist

- [x] My project includes a `README.md` file with a brief overview of the project, how to set it up, and any relevant information.
- [x] I have added a screenshot or screen recording showing the functionality of the app.
- [x] I have tested the project and confirmed it works as expected.

## Screenshots

<p>
   
  
  

</p>

## Project README

Please ensure that your project's `README.md` is detailed and includes:

- A brief introduction to the project.
- Installation steps.
- Key features.
- Any additional instructions or information.

## Additional Notes

Is there anything else the maintainers should know about your contribution?

---

Thank you for your contribution!

### CODE REVIEW

File: Mani-Quotify/app/src/main/java/com/mani/quotify007/MainActivity.kt [hunk 1]
```diff
@@ -0,0 +1,180 @@
...
+                    )
+                }
+            }
+        }
+    }
+}
+
+@Preview(showBackground = true)
+@Composable
+fun GreetingPreview() {
```
Comment (anandwana001) [inline Mani-Quotify/app/src/main/java/com/mani/quotify007/MainActivity.kt]: the function name is not matching to its responsibility

File: Mani-Quotify/app/src/main/java/com/mani/quotify007/MainActivity.kt [hunk 2]
```diff
@@ -0,0 +1,180 @@
...
+            NavigationGraph(
+                navController = navController,
+                modifier = Modifier.padding(innerPadding)
+            )
+        }
+    }
+}
+
+@Composable
+fun BottomAppBar(navController: NavHostController) {
```
Comment (anandwana001) [inline Mani-Quotify/app/src/main/java/com/mani/quotify007/MainActivity.kt]: why everything is under the same file?

File: Mani-Quotify/app/src/main/java/com/mani/quotify007/MainActivity.kt [hunk 3]
```diff
@@ -0,0 +1,180 @@
...
+@Composable
+fun HomeScreen() {
+    Column(
+        modifier = Modifier
+            .fillMaxSize()
+            .padding(16.dp),
+        verticalArrangement = Arrangement.Center,
+        horizontalAlignment = Alignment.CenterHorizontally
+    ) {
+        Text("Quote of the Day", fontStyle = FontStyle.Italic, fontSize = 30.sp)
```
Comment (anandwana001) [inline Mani-Quotify/app/src/main/java/com/mani/quotify007/MainActivity.kt]: why hardcoded estring?

File: Mani-Quotify/app/src/main/java/com/mani/quotify007/MainActivity.kt [hunk 4]
```diff
@@ -0,0 +1,180 @@
...
+        Box(
+            modifier = Modifier
+                .fillMaxWidth()
+                .height(250.dp)
+                .padding(top = 16.dp, start = 16.dp, end = 16.dp)
+                .clip(MaterialTheme.shapes.medium)
+                .background(
+                    //GradientStart: “#D9FF43”, GradientCenter: “#F67831”, GradientEnd: “#FF1493”
+                    //generateRandomGradient()
```
Comment (anandwana001) [inline Mani-Quotify/app/src/main/java/com/mani/quotify007/MainActivity.kt]: do ew need this?

**Review by anandwana001 — COMMENTED** [review]
can we add relevant screenshots in PR description and also a working video of the app?

**Review by anandwana001 — CHANGES_REQUESTED** [review]
few comments

File: Mani-Quotify/app/src/main/java/com/mani/quotify007/MainActivity.kt (hunk 1 above)
Comment (itmani1991) [reply Mani-Quotify/app/src/main/java/com/mani/quotify007/MainActivity.kt]: Resolved

File: Mani-Quotify/app/src/main/java/com/mani/quotify007/MainActivity.kt (hunk 2 above)
Comment (itmani1991) [reply Mani-Quotify/app/src/main/java/com/mani/quotify007/MainActivity.kt]: Resolved

File: Mani-Quotify/app/src/main/java/com/mani/quotify007/MainActivity.kt (hunk 3 above)
Comment (itmani1991) [reply Mani-Quotify/app/src/main/java/com/mani/quotify007/MainActivity.kt]: Resolved

File: Mani-Quotify/app/src/main/java/com/mani/quotify007/MainActivity.kt (hunk 4 above)
Comment (itmani1991) [reply Mani-Quotify/app/src/main/java/com/mani/quotify007/MainActivity.kt]: Resolved

**Comment by itmani1991** [comment]
Comments Addressed, Please review.

File: Mani-Quotify/app/src/main/java/com/mani/quotify007/view/FavoritesScreen.kt [hunk 5]
```diff
@@ -0,0 +1,79 @@
...
+fun FavoritesScreen() {
+    val favoriteQuotes = remember { mutableStateListOf(
+        "The best way to predict the future is to invent it.",
+        "Life is 10% what happens to us and 90% how we react to it.",
+        "The only way to do great work is to love what you do."
+    ) }
+
+    fun removeFavorite(quote: String) {
+        favoriteQuotes.remove(quote)
+    }
```
Comment (anandwana001) [inline Mani-Quotify/app/src/main/java/com/mani/quotify007/view/FavoritesScreen.kt]: Any rational behind nested function?

File: Mani-Quotify/app/src/main/java/com/mani/quotify007/view/FavoritesScreen.kt [hunk 6]
```diff
@@ -0,0 +1,79 @@
...
+
+    Column(
+        modifier = Modifier
+            .fillMaxSize()
+            .padding(16.dp),
+        verticalArrangement = Arrangement.Top,
```
Comment (anandwana001) [inline Mani-Quotify/app/src/main/java/com/mani/quotify007/view/FavoritesScreen.kt]: isn't the top is default value? why to pass?

File: Mani-Quotify/app/src/main/java/com/mani/quotify007/view/utils/BottomNavItem.kt [hunk 7]
```diff
@@ -0,0 +1,13 @@
+package com.mani.quotify007.view.utils
+
+import androidx.compose.material.icons.Icons
+import androidx.compose.material.icons.filled.Favorite
+import androidx.compose.material.icons.filled.Home
+import androidx.compose.material.icons.filled.Search
+import androidx.compose.ui.graphics.vector.ImageVector
+
+sealed class BottomNavItem(var title: String, var icon: ImageVector, var route: String) {
+    data object Home : BottomNavItem("Home", Icons.Default.Home, "home")
```
Comment (anandwana001) [inline Mani-Quotify/app/src/main/java/com/mani/quotify007/view/utils/BottomNavItem.kt]: data class vs data object?

sort of interview question

File: Mani-Quotify/app/src/main/java/com/mani/quotify007/view/utils/StringConstants.kt [hunk 8]
```diff
@@ -0,0 +1,8 @@
+package com.mani.quotify007.view.utils
+
+object StringConstants {
```
Comment (anandwana001) [inline Mani-Quotify/app/src/main/java/com/mani/quotify007/view/utils/StringConstants.kt]: why object is required here?

**Review by anandwana001 — CHANGES_REQUESTED** [review]

File: Mani-Quotify/app/src/main/java/com/mani/quotify007/view/utils/BottomNavItem.kt (hunk 7 above)
Comment (itmani1991) [reply Mani-Quotify/app/src/main/java/com/mani/quotify007/view/utils/BottomNavItem.kt]: data class - if we need multiple instances with different data.
data object -if we need a single instance with fixed data.

File: Mani-Quotify/app/src/main/java/com/mani/quotify007/view/FavoritesScreen.kt (hunk 5 above)
Comment (itmani1991) [reply Mani-Quotify/app/src/main/java/com/mani/quotify007/view/FavoritesScreen.kt]: Resolved

File: Mani-Quotify/app/src/main/java/com/mani/quotify007/view/FavoritesScreen.kt (hunk 6 above)
Comment (itmani1991) [reply Mani-Quotify/app/src/main/java/com/mani/quotify007/view/FavoritesScreen.kt]: Resolved

File: Mani-Quotify/app/src/main/java/com/mani/quotify007/view/utils/StringConstants.kt (hunk 8 above)
Comment (itmani1991) [reply Mani-Quotify/app/src/main/java/com/mani/quotify007/view/utils/StringConstants.kt]: Resolved

**Comment by itmani1991** [comment]
Please review.

**Comment by anandwana001** [comment]
can you add screenshots?
@itmani1991

**Comment by itmani1991** [comment]
@anandwana001 Added screenshots

**Pull request merged**